### ISSUE TITLE
Can TUNIT do few-shot generation?

### ISSUE THREAD

**Issue by ammar-deep** [clovaai/tunit#29]
I was wondering that have you tried using TUNIT for the few-shot generation task like FUNIT? 

Given few images of an unseen class, can TUNIT transfer its style (from the guiding network) to a seen content during inference? 
OR
it will require fine-tuning the network on the new class?

**Comment by FriedRonaldo** [comment]
Unfortunately, it can not be directly applied to few-shot setting because we do not design the model as a few-shot model in terms of the objective functions.

You can refer this issue (#2 , https://github.com/clovaai/tunit/issues/2#issuecomment-645234042).

**Comment by ammar-deep** [comment]
> Unfortunately, it can not be directly applied to few-shot setting because we do not design the model as a few-shot model in terms of the objective functions.

As you mentioned you did not design the loss function for the few-shot task. By this do you mean that the multi-task discriminator of TUNIT is conditioned on domain rather than the style class (like FUNIT)? or the feature matching loss used in FUNIT? 

> You can refer this issue (#2 , [#2 (comment)](https://github.com/clovaai/tunit/issues/2#issuecomment-645234042)).

Thanks Ill look into it.

**Comment by FriedRonaldo** [comment]
We define the style as a domain, therefore, using multi-task discriminator might be maintained. 

The objectives of each work are:
FUNIT adopted i) GAN loss ii) feature matching loss iii) image reconstruction loss
TUNIT adopted i) GAN loss ii) style loss iii) image reconstruction loss
So, you can try changing the style loss of TUNIT into feature matching loss for few-shot generation.

However, this simple modification might not work because TUNIT does not aim few-shot generation.

**Comment by ammar-deep** [comment]
Yes seems like replacing the style loss with feature matching loss will not work so well. 

My intuition behind this is that the guiding network is mainly responsible for injecting the style code into the generator. However, for this particular case (reference images of unseen domain) the guiding network will generate a style code that will represent the closest learnt embeddings (style codes similar to the training data) which will not represent the style of the unseen domain.
